packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -406,14 +402,45 @@ export async function getLinkRemovalByDidExtrinsic(
 /**
  * Return the default sign callback, which uses the address argument to crete a signing closure for the given payload.
  *
- * @param keyring The keyring to retrieve the signing key.
+ * @param keypair The keypair to sign the data with.
  * @returns The signature generating callback that uses the keyring to sign the input payload using the input address.
  */
-export function defaultSignerCallback(keyring: Keyring): LinkingSignerCallback {
-  return (payload: HexString, address: Address): Promise<HexString> =>
-    Promise.resolve(
-      u8aToHex(keyring.getPair(address).sign(payload, { withType: false }))
-    )
+export function makeLinkingSignCallback(
+  keypair: KeyringPair
+): LinkingSignCallback {
+  return async function sign(payload: HexString): Promise<Uint8Array> {
+    return keypair.sign(payload, { withType: false })
+  }
+}
+
+function getUnprefixedSignature(
```

Comment:
Extracting this into function allows to more clearly communicate the intent and to have a neater code

Reply:
that's true! Tbh I still don't know 100% if this check is actually required or if the chain would also accept the prefix. Was hard to find out and we were pressed on time so we just left it. Maybe @trusch could try to figure that out?